core/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -621,6 +544,114 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         }
         return builder;
     }
+    public static Settings.Builder fromXContent(XContentParser parser, Settings.Builder builder) throws IOException {
+        return fromXContent(parser, builder, true);
+    }
+
+    public static Settings.Builder fromXContent(XContentParser parser, Settings.Builder builder, boolean allowNullValues)
+        throws IOException {
+        if (parser.currentToken() == null) {
+            parser.nextToken();
+           // throw new ElasticsearchParseException("malformed, expected settings to start with 'object', instead was [{}]", token);
+        }
+        if (parser.currentToken() != XContentParser.Token.START_OBJECT) {
```

Comment:
This can be replaced by 
` ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);`  from `XContentParserUtils`

Reply:
Looks like the toXContent() and fromXContent() are not completely mirrored; the former does not render any root object while the latter expects it.

Reply:
I am not going to change the mirroring in this pr it's complex enough I think.